### PR TITLE
Remove Flash and change DOM module

### PR DESCRIPTION
**Pull request by BrandonOCasey**
## Description

Remove the flash tech and tests from video.js. This requires a major version bump in video.js.
## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by gkatsev** [comment]
Thoughts on re-adding Flash as a default plugin right now and then we can remove that plugin in december when we do the major version bump?

**Comment by BrandonOCasey** [comment]
I think we may want to hold off on this until we can do a major version bump, seems like it will be a lot of work to resolve the circular dependencies here. (flash needs video.js to be loaded so that it can load itself, but video.js wants to include flash during load... etc)

File: src/js/video.js [hunk 1]
```diff
@@ -719,6 +721,18 @@ videojs.appendContent = Dom.appendContent;
  */
 videojs.insertContent = Dom.insertContent;
 
+/**
+ * Export the Dom utilities for use in external plugins
+ * and Tech's
+ */
+videojs.Dom = Dom;
```
Comment (misteroneill) [inline src/js/video.js]: Looks like this is in here twice (also at L567).

**Comment by BrandonOCasey** [comment]
update for the community. This is going to wait until `video.js` 6.0 due to some technical limitations, look for it in the next major release.

File: src/js/video.js (hunk 1 above)
Comment (misteroneill) [inline src/js/video.js]: Also, can we expose it as `dom` instead? It's not a constructor; so, the casing doesn't make much sense.

File: src/js/video.js [hunk 2]
```diff
@@ -719,6 +721,18 @@ videojs.appendContent = Dom.appendContent;
...
+
+/**
+ * Export the Url utilities for use in external plugins
+ * and Tech's
+ */
+videojs.Url = Url;
```
Comment (misteroneill) [inline src/js/video.js]: Can we expose this as `url` instead? It's not a constructor; so, the casing doesn't make much sense.

File: src/js/video.js [hunk 3]
```diff
@@ -564,6 +564,8 @@ videojs.AudioTrack = AudioTrack;
  */
 videojs.VideoTrack = VideoTrack;
 
+videojs.Dom = Dom;
```
Comment (misteroneill) [inline src/js/video.js]: This can be removed.

**Comment by BrandonOCasey** [comment]
pushed upstream to get tests running on browserstack see #3956

**Pull request closed without merging**